src/app/rpmostree-compose-builtin-jigdo2commit.c

```diff
@@ -0,0 +1,365 @@
+  return TRUE;
+}
+
+static DnfPackage *
+query_nevra (DnfContext *dnfctx,
+             const char *name,
+             guint64     epoch,
+             const char *version,
+             const char *release,
+             const char *arch,
+             GError    **error)
+{
+  hy_autoquery HyQuery query = hy_query_create (dnf_context_get_sack (dnfctx));
+  hy_query_filter (query, HY_PKG_NAME, HY_EQ, name);
+  hy_query_filter_num (query, HY_PKG_EPOCH, HY_EQ, epoch);
+  hy_query_filter (query, HY_PKG_VERSION, HY_EQ, version);
+  hy_query_filter (query, HY_PKG_RELEASE, HY_EQ, release);
+  hy_query_filter (query, HY_PKG_ARCH, HY_EQ, arch);
+  g_autoptr(GPtrArray) pkglist = hy_query_run (query);
+  if (pkglist->len == 0)
+    return glnx_null_throw (error, "Failed to find package '%s'", name);
+  return g_object_ref (pkglist->pdata[0]);
+}
+
+static gboolean
+commit_and_print (RpmOstreeJigdo2CommitContext *self,
+                  GCancellable                 *cancellable,
+                  GError                      **error)
+{
+  OstreeRepoTransactionStats stats;
+  if (!ostree_repo_commit_transaction (self->repo, &stats, cancellable, error))
+    return FALSE;
+
+  g_print ("Metadata Total: %u\n", stats.metadata_objects_total);
+  g_print ("Metadata Written: %u\n", stats.metadata_objects_written);
+  g_print ("Content Total: %u\n", stats.content_objects_total);
+  g_print ("Content Written: %u\n", stats.content_objects_written);
+  g_print ("Content Bytes Written: %" G_GUINT64_FORMAT "\n", stats.content_bytes_written);
+
+  return TRUE;
+}
+
+static int
+compare_pkgs_reverse (gconstpointer ap,
+                      gconstpointer bp)
+{
+  DnfPackage **a = (gpointer)ap;
+  DnfPackage **b = (gpointer)bp;
+  return dnf_package_cmp (*b, *a); // Reverse
+}
+
+static gboolean
+impl_jigdo2commit (RpmOstreeJigdo2CommitContext *self,
+                   const char                   *oirpm_id,
+                   GCancellable                 *cancellable,
+                   GError                      **error)
+{
+  g_autoptr(GKeyFile) tsk = g_key_file_new ();
+
+  if (opt_releasever)
+    g_key_file_set_string (tsk, "tree", "releasever", opt_releasever);
+  if (opt_enable_rpmmdrepo)
+    g_key_file_set_string_list (tsk, "tree", "repos",
+                                (const char *const*)opt_enable_rpmmdrepo,
+                                g_strv_length (opt_enable_rpmmdrepo));
+  g_autoptr(RpmOstreeTreespec) treespec = rpmostree_treespec_new_from_keyfile (tsk, error);
+  if (!treespec)
+    return FALSE;
+
+
+  if (!rpmostree_context_setup (self->ctx, NULL, NULL, treespec, cancellable, error))
+    return FALSE;
+  if (!rpmostree_context_download_metadata (self->ctx, cancellable, error))
+    return FALSE;
+
+  DnfContext *dnfctx = rpmostree_context_get_dnf (self->ctx);
+  g_autoptr(DnfPackage) oirpm_pkg = NULL;
+  { hy_autoquery HyQuery query = hy_query_create (dnf_context_get_sack (dnfctx));
+    const char *at = strchr (oirpm_id, '@');
```

Comment:
Hmm, can we make this a separate `--version` argument or even just a second positional arg?